gmso/external/convert_hoomd.py

```diff
@@ -0,0 +1,881 @@
+        lennard_jones_potential,
+        harmonic_bond_potential,
+        harmonic_angle_potential,
+        periodic_torsion_potential,
+        opls_torsion_potential,
+        rb_torsion_potential,
+    ]
+    potential_types = check_compatibility(top, accepted_potentials)
+    return potential_types
+
+
+def _parse_nonbonded_forces(
+    top, nlist_buffer, potential_types, potential_refs, base_units
+):
+    """Parse nonbonded forces."""
+    # Set up helper methods to parse different nonbonded forces.
+    def _parse_lj(container, atypes, combining_rule):
+        """Parse LJ forces."""
+        for atype1, atype2 in itertools.combinations_with_replacement(
+            atypes, 2
+        ):
+            comb_epsilon = statistics.geometric_mean(
+                [atype1.parameters["epsilon"], atype2.parameters["epsilon"]]
+            )
+            if top.combining_rule == "lorentz":
+                comb_sigma = np.mean(
+                    [atype1.parameters["sigma"], atype2.parameters["sigma"]]
+                )
+            elif top.combining_rule == "geometric":
+                comb_sigma = statistics.geometric_mean(
+                    [atype1.parameters["sigma"], atype2.parameters["sigma"]]
+                )
+            else:
+                raise ValueError(
+                    f"Invalid combining rule provided ({combining_rule})"
+                )
+
+            container.params[(atype1.name, atype2.name)] = {
+                "sigma": comb_sigma,
+                "epsilon": comb_epsilon,
+            }
+
+        return container
+
+    def _parse_buckingham(container, atypes):
+        return None
+
+    def _parse_lj0804(container, atypes):
+        return None
+
+    def _parse_lj1208(container, atypes):
+        return None
+
+    def _parse_mie(container, atypes):
+        return None
+
+    unique_atypes = top.atom_types(filter_by=PotentialFilters.UNIQUE_NAME_CLASS)
+
+    # Grouping atomtype by group name
+    groups = dict()
+    for atype in unique_atypes:
+        group = potential_types[atype]
+        if group not in groups:
+            groups[group] = [atype]
+        else:
+            groups[group].append(atype)
+
+    # Perform units conversion based on the provided base_units
+    for group in groups:
+        expected_units_dim = potential_refs[group][
+            "expected_parameters_dimensions"
+        ]
+        groups[group] = _convert_params_units(
+            groups[group], expected_units_dim, base_units
+        )
+
+    atype_group_map = {
+        "LennardJonesPotential": {
+            "container": hoomd.md.pair.LJ,
+            "parser": _parse_lj,
+        },
+        "BuckinghamPotential": {
+            "container": hoomd.md.pair.Buckingham,
+            "parser": _parse_buckingham,
+        },
+        "MiePotential": {"container": hoomd.md.pair.Mie, "parser": _parse_mie},
+    }
+
+    nlist = hoomd.md.nlist.Cell(exclusions=["bond", "1-3"], buffer=nlist_buffer)
```

Comment:
Do we want to include the option to add "1-4" to the nlist exclusions? IMO that would be useful.

Reply:
Now included (dynamically added from top.scaling_factors)